x/validators/handler.go

```diff
@@ -92,20 +92,20 @@ func (h updateHandler) validate(ctx weave.Context, store weave.KVStore, tx weave
 DiffLoop:
 	for _, v := range diff {
 		for key, validator := range validatorSlice {
-			if bytes.Equal(v.PubKey.Data, validator.Pubkey.Data) {
+			if bytes.Equal(v.PubKey.Data, validator.PubKey.Data) {
```

Comment:
I think this whole DiffLoop would be a possible weave-level function.

eg. `AlreadyHasUpdate(validatorSlice, v)`